src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -95,9 +95,21 @@ export function IndexPatternProvider(Private, Notifier, config, kbnIndex, Promis
     return promise;
   }
 
-  function containsFieldCapabilities(fields) {
-    return _.any(fields, (field) => {
-      return _.has(field, 'aggregatable') && _.has(field, 'searchable');
+  function isFieldRefreshRequired(indexPattern) {
+    if (!indexPattern.fields) {
+      return true;
+    }
+
+    return indexPattern.fields.every(field => {
+      // added in 5.0 #8421
```

Comment:
I'm pretty sure there are a couple places like this. The main reason I would but these types of comments in is that these lines are here for backwards compatibility, so if we want to remove/edit them we need to consider when they were added. It's not a ton different than most things, but having a note for why this somewhat strange series of checks exists is nothing but helpful if you ask me.